core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -474,6 +477,7 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws
         if (this.termsLookup != null) {
             TermsLookup termsLookup = new TermsLookup(this.termsLookup);
             if (termsLookup.index() == null) { // TODO this should go away?
+                DEPRECATION_LOGGER.deprecated("Omitting the index in terms lookup is deprecated");
```

Comment:
should we rather issue the deprecation warning once in the coord node while parsing?